sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,252 @@
+#title          :discretization.py
+#description    :Implementation of a few discretization algorithms
+#author         :Henry Lin <hlin117@gmail.com>
+#version        :0.0.1
+#python_version :2.7.6
+#==============================================================================
+
+from collections import Counter
+from math import log
+import numpy as np
+from itertools import izip, islice
+
+
+def log2(x):
+    return log(x, 2) if x > 0 else 0
+
+
+class MDLP(object):
```

Comment:
This is not in accordance with scikit-learn's naming scheme. From a glance at your code, it is applicable to multiclass classification problems only. Perhaps we should call this `ClassificationDiscretizer` or `MDLDiscretizer`.